### PR TITLE
Implement KNNRegressor

### PR DESCRIPTION
**Pull request by bojanbla**
Summary:

- Implement multi variable linear regression using **KNN classifier**
- Short description added in **README.md**

### CODE REVIEW

File: ex-02-multi-var-lin-reg/knn_regressor.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+        :param x: input vector of features for one point
+        :param y: input vector of features for one point
+        :return: euclidean distance between two points
+        """
+        distance = 0
+        for i in range(len(x)):
+            distance += (x[i] - y[i]) ** 2
+
+        return sqrt(distance)
```
Comment (drmarjanovic) [inline ex-02-multi-var-lin-reg/knn_regressor.py]: You can use **map** with **zip** here.

File: ex-02-multi-var-lin-reg/knn_regressor.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        """
+        Function that calculate manhattan distance between two points
+
+        :param x: input vector of features for one point
+        :param y: input vector of features for one point
+        :return: manhattan distance between two points
+        """
+        distance = 0
+        for i in range(len(x)):
+            distance += abs(x[i] - y[i])
```
Comment (drmarjanovic) [inline ex-02-multi-var-lin-reg/knn_regressor.py]: Also, you can use **map** and **zip** here. 😄

File: ex-02-multi-var-lin-reg/test.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+from util import get_arg_or_else, hello
+
+if __name__ == '__main__':
+    test_path = get_arg_or_else(1, "data/test.csv")
+    pd.options.mode.chained_assignment = None
+
+    hello()
+
+    # Read training data from file
+    df = pd.read_csv('data/train.csv')
```
Comment (drmarjanovic) [inline ex-02-multi-var-lin-reg/test.py:15]: Use more clear name for variables. What does this mean? 😄

File: ex-02-multi-var-lin-reg/test.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    feature_normalization(data, binary={'sex': ['F', 'M'], 'schoolsup': ['yes', 'no'], 'higher': ['yes', 'no']},
+                          nominal={'reason': ['reputation', 'home', 'course', 'other']}, norm_params=norm_params)
+
+    # Separate data to validation and train
+    split_value = int(len(data) * 0.2)
+    validation, train = data.ix[:split_value, :], data.ix[split_value:, :]
+
+    x_train, y_train = train.ix[:, :-1], train.ix[:, -1]
+    x_validation, y_validation = train.ix[:, :-1], train.ix[:, -1]
+
```
Comment (drmarjanovic) [inline ex-02-multi-var-lin-reg/test.py]: I'm not sure that we now need to split data. Validation set will be provided from his side. Maybe, you can make additional parameter in **args** for this. Maybe we should use all this provided data for training.

**Review by drmarjanovic — CHANGES_REQUESTED** [review]

File: ex-02-multi-var-lin-reg/test.py (hunk 3 above)
Comment (stkeky) [reply ex-02-multi-var-lin-reg/test.py:15]: `df` is 'standard' short-name for `pandas.DataFrame` so it's not that bad 😃

File: ex-02-multi-var-lin-reg/knn_regressor.py (hunk 1 above)
Comment (stkeky) [reply ex-02-multi-var-lin-reg/knn_regressor.py]: Change was made to use `map` and `reduce`.

File: ex-02-multi-var-lin-reg/knn_regressor.py (hunk 2 above)
Comment (stkeky) [reply ex-02-multi-var-lin-reg/knn_regressor.py]: Change was made to use `map` and `reduce`.

**Comment by stkeky** [comment]
I added 2 more fields (`internet` & `guardian`) which only in this combination gives **`validation_rmse = 2.28`** and **`test_rmse = 2.76`**.

 _RMSEs were **2.43** and **2.91** respectfully._

File: ex-02-multi-var-lin-reg/test.py (hunk 4 above)
Comment (stkeky) [reply ex-02-multi-var-lin-reg/test.py]: I agree, TA provided separate `*.csv` files.

**Review by drmarjanovic — APPROVED** [review]

**Pull request merged**